reference/interbit-ui-components/CONTRIBUTING.adoc

```diff
@@ -0,0 +1,187 @@
+
+| `src/styleguide/`
+| Build files for React Styleguidist.
+
+| `index.js`
+| Export statements for components.
+
+| `styleguide.config.js`
+| Configuration file for React Styleguidist.
+
+|===
+
+
+== Creating components
+
+=== Code structure
+
+We are in the process of defining the code structure for
+Interbit UI components. At this time, we ask that contributing developers use
+the existing code as an example of how to structure your component's code.
+
+Simple, presentational components should go in the `src/components/UIKit`
+directory. More complex components should be in their own directory within
+`src/components`. Each component must be exported from `src/index.js`.
+
+=== SCSS
+
+For each component:
+
+- The styles should be in their own file, say `_my-component.scss`, in the
+`src/scss/interbit` directory. This file should be imported in
+`src/scss/interbit.scss`.
+- The styles should be contained in a class named after the component and
+prefixed with `ibweb-`. For example, `MyComponent` should have a top-level
+class called `ibweb-my-component`. See `src/scss/interbit/_footer.scss` for how
+styles should be nested.
+- The CSS must be supported by the latest versions of Chrome, Firefox, Safari,
+and Edge. If in doubt, see link:https://caniuse.com/[caniuse.com].
+
+=== Image assets
+
+Image assets such as icons and illustrations should be placed in `src/assets` and should satisfy the following:
+
+- Icons files are placed in `src/assets/icons`.
+- Image asset files are in `svg` format wherever possible.
+- Files should not exceed 25KB in size at this time.
+
+=== Tests
+
+Components must have an export test in `src/tests/exports.test.js` for each
+(sub)-component exported from `src/index.js`. This verifies that the component
+is in fact exported, and is usable externally.
+
+UI Kit components must have, at the minimum, rendering tests in
+`src/tests/uiKit.test.js`. Interbit UI components uses Jest and Enzyme for
+testing and rendering components.
+
+=== Documentation
+
+Interbit pulls documentation out of JSDoc style code comments. For each
+component:
+
+- Give a brief description of the component before the class declaration.
+- Give a description of each prop that the component accepts
+
+For example, the `LinkBar` component has the following code comments.
+
+[source,js]
+----
+{% include "linkBar.js" %}
+----
+
+Each UI Kit component should be accompanied by an example file,
+`MyComponent.md`. link:https://react-styleguidist.js.org/docs/documenting.html[React Styleguidist]
+renders the components and JSDoc comments in a style guide.
+
+The example file should include:
+
+- An example of the rendered component.
+- An example of each variant of the component. For instance, we might apply
+`className="blue"` to a button component to render a blue button.
+
+The following is `LinkBar`'s example `md` file:
+
+[source,md]
+----
+Full-sized examples. To see how the component behaves responsively, change your
```

Comment:
Extract the Markdown example into a separate file, probably named 'component-example.markdown'. One of the doc CI checks looks for `*.md` files in the rendered HTML folder to identify files not included in `SUMMARY.md`, so the example file can't have the `.md` extension.

Reply:
Thanks, I wanted to pull it out into a separate file but it was squawking at me.